sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/README.md

```diff
@@ -8,112 +8,313 @@ The authentication events trigger for Azure Functions allows you to implement a
+1. In the top menu of Visual Studio, select **Project**, then **Manage NuGet packages**.
+2. Select the **Browse** tab, then search for and select *Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents* in the right pane. Select **Install**.
+3. Apply and accept the changes in the popups that appear.
 
-If the `accessTokenAcceptedVersion` property is set to `1` or `null`:
-6. Set the `Issuer URL to "https://sts.windows.net/{tenantId}/"
-7. Set an 'Allowed Audience' to the Application ID URI (also known as`identifierUri`). It should be in the format of`api://{azureFunctionAppName}.azurewebsites.net/{resourceApiAppId}` or `api://{FunctionAppFullyQualifiedDomainName}/{resourceApiAppId}` if using a [custom domain name](https://learn.microsoft.com/azure/dns/dns-custom-domain#:~:text=Azure%20Function%20App%201%20Navigate%20to%20Function%20App,Custom%20domain%20text%20field%20and%20select%20Validate.%20).
+#### [Visual Studio Code](#tab/visual-studio-code)
 
-By default, the Authentication event trigger will validate that Azure Function authentication integration is configured and it will check that the **client** in the token is set to `99045fe1-7639-4a75-9d4a-577b6ca3810f` (via the `azp` or `appid` claims in the token).
+1. Open the **Terminal** in Visual Studio Code, and navigate to the project folder.
+2. Enter the following command into the console to install the *Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents* NuGet package.
 
-If you want to test your API against some other client that is not Microsoft Entra authentication events service, like using Postman, you can configure an _optional_ application setting:
+```console
+dotnet add package Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
+```
 
-* **AuthenticationEvents__CustomCallerAppId** - the guid of your desired client. If not provided, `99045fe1-7639-4a75-9d4a-577b6ca3810f` is assumed.
+---
 
-#### Have the trigger validate the token
+### Add the sample code
 
-In local environments or environments that aren't hosted in the Azure Function service, the trigger can do the token validation. Set the following application settings in the [local.settings.json](https://learn.microsoft.com/azure/azure-functions/functions-develop-local#local-settings-file) file:
+The function API is the source of extra claims for your token. For the purposes of this article, we're hardcoding the values for the sample app. In production, you can fetch information about the user from external data store.
 
-* **AuthenticationEvents__TenantId** - your tenant ID
-* **AuthenticationEvents__AudienceAppId** - the same value as "Allowed audience" in option 1.
-* **AuthenticationEvents__CustomCallerAppId** (_optional_) - the guid of your desired client. If not provided, `99045fe1-7639-4a75-9d4a-577b6ca3810f` is assumed.
+In your *AuthEventsTrigger.cs* file, replace the entire contents of the file with the following code:
 
-An example `local.settings.json` file:
+```cs
```

Comment:
All code in samples and readmes in this repository are required to be [snippets](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets).

We don't write end-to-end classes here.  The goal is to identify key areas that customers would be looking for to make things work and including a focused snippet that they can copy/paste and use.   These should be self-contained and not require other snippets and they should be free of boilerplate code unrelated to a specific scenario.

Reply:
Reduced this to a snippet

Reply:
This needs to be a snippet, which is actual test code that is injected.   This allows us to ensure that it does not rot and become uncompilable.   Details are in the link from my comment.

Reply:
Updated by creating a cs file and adding `#region Snippet` to them and referencing them